ENIGMAsystem/SHELL/Universal_System/Extensions/Asynchronous/ASYNCdialog.h

```diff
@@ -27,7 +27,9 @@ namespace enigma_user {
   int show_message_async(string str);
   int show_question_async(string str);
   int get_string_async(string message, string def, string cap="");
-  int get_integer_async(string message, string def, string cap="");
+  int get_password_async(string message, string def, string cap="");
+  int get_integer_async(string message, double def, string cap="");
```

Comment:
@RobertBColton please explain to Josh why this is. Apparently Josh doesn't know how the async functions work. If he knew that much, he would understand why the argument is a double.

Reply:
Erm..... jesus I hate YoYoGames. It should technically be int @JoshDreamland but for GM compatibility we need it to be a double. The function has been this way since GM5. Further, `get_integer_async` should just match the signature of `get_integer` for the most part. And in both cases they have to be double for GM compatibility.

Reply:
@RobertBColton well technically rather, the issue isn't that they gave the function the wrong return type, they should've given it the proper name, where as get_number() would've been much more appropriate.

Reply:
Right, that's the stinker, `get_number` would have been better for var and then `get_integer` would have made sense for a correctly typed int (meaning the dialog would also making float invalid/etc offer validation). Stupid on their part.

Reply:
@time-killer-games regardless of whether you go with int or double here for `get_integer`, you need to make the return type match the type of `def` either way. Don't make `def` double and then leave the return type as integer. That match should at least be consistent.

Reply:
@RobertBColton I thought I was already doing that. Remember, this is the async function we are looking at, the return value isn't what we will be retrieving from the data structure, but rather, whatever double the user entered in the textbox

Reply:
Oh that's right, I'm sorry, `get_integer_async` returns the message id. Ok, yeah that's fine then.